### ISSUE TITLE
Intro to ARIA states

### ISSUE THREAD

**Issue by nirjan-dev** [nirjan-dev/learning-to-code#62]
[link to the article](https://www.a11ywithlindsey.com/blog/introduction-aria-states)

# The aria-expanded attribute

Anytime we hide content that expands upon user action, use aria-expanded. When you use JavaScript to toggle it, it becomes a better user experience.

# The hidden attribute

aria-hidden only hides elements from screen readers, while hidden hides from everyone.
I ask myself if there are HTML elements that are primarily for decoration. If so, I might want to use aria-hidden to hide it from screen readers but not from visually-abled users.

One of my favorite reason’s to use aria-hidden is to prevent repetition. This, for example, is the Home button for Twitter:

```
<a class="js-nav js-tooltip js-dynamic-tooltip">
  <span class="Icon Icon--home Icon--large"></span>
  <span class="Icon Icon--homeFilled Icon--large u-textUserColor"></span>
  <span class="text" aria-hidden="true">Home</span>
  <span class="u-hiddenVisually a11y-inactive-page-text">Home</span>
  <span class="u-hiddenVisually a11y-active-page-text">
    Home, current page.
  </span>
  <span class="u-hiddenVisually hidden-new-items-text">
    New Tweets available.
  </span>
</a>
```
In this instance, the screen reader announces one of the “a11y page texts” based on the context of the page. I want you to take note of this line of code:
```
<span class="text" aria-hidden="true">Home</span>
```
This is the one that is actually on the screen. We have the dynamic options for the screen reader like “Home,” “Home, current page,” and “Load new Tweets.” We don’t need that context on the Home button if we can see the visual cues.

With that being said, my rule is if we are toggling states, I use hidden. If there’s duplicate or unnecessary content, I’ll use aria-hidden.

# The aria-current attribute

eaning (with an exception) 0 of the styling is exposed to screen readers. Using aria-current helps ensure that we communicate the context to screen reader users. According to WAI-ARIA 1.1, there are a few values that aria-current could take on:

    page for a link within a set of pagination links, the current page as represented in the navigation, etc.
    step for a step-based process.
    date for a current date.
    time for a current time.

Additionally, this should be only one element in a set of elements can have an aria-current value. Otherwise, it will confuse the screen reader!
